codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/rustlang/RustReservedWords.kt

```diff
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-package software.amazon.smithy.rust.codegen.client.rustlang
+package software.amazon.smithy.rust.codegen.core.rustlang
```

Comment:
I think `RustReservedWords` should stay in client for now. It needs to be broken up into multiple composed classes since right now it has client-specific logic in it to handle the `Unknown` enum variants.

Reply:
I agree we should refactor that bit out. However, I can't keep it in `codegen-client`, since `BuilderGenerator`, that is `codegen-core` depends on it, and we can't have `core` depending on `client`. We should eventually put `BuilderGenerator` in `codegen-client` anyway, since in #1342 the server project has its own builder generators. However, if I do so now in this PR, several things break, because the extension functions defined in `BuilderGenerator.kt` (`StructureShape.builderSymbol`, `RuntimeConfig.operationBuildError()`, `MemberShape.setterName()`...) are ingrained throughout `codegen-core`. Untangling all these is related to the `CodegenTarget` and reducing branching cleanup, that we should tackle separately.

Reply:
I opened https://github.com/awslabs/smithy-rs/issues/1766 to track this.